### DISCUSSION TITLE
Issue rendering multiple nested array objects

### DISCUSSION THREAD

**Discussion by barbigrade** [neuefische/questions#181] [Web] [answered]
Hi all,

A silly question regarding mapping and rendering nested array objects from a JSON file, hope someone can point me in the right direction at least.

I am trying to render cards for each [id] page, consisting of the products that belong to each bakery menu (see [user story](https://github.com/barbigrade/capstone-project/issues/4) and JSON structure below) but I'm not sure how to loop over each product and make sure that each bakery page renders out the corresponding menu. Currently, no data is being rendered apart from the first component displayed on each page: 
<img width="400" alt="Screenshot 2022-09-28 at 16 05 34" src="https://user-images.githubusercontent.com/107254671/192800327-0b3ecc0f-4243-45e3-8fcc-e10ba912f6ea.png">

The function _getAllProductMenus()_ returns all the _productMenu_ arrays - these were passed down as props. 
_bakeryData_ returns all data from the file.
For more context: 

JSON:
<img width="400" alt="Screenshot 2022-09-28 at 15 39 40" src="https://user-images.githubusercontent.com/107254671/192793938-79e85feb-aa40-404a-b8a4-4e13d450e5b9.png">

The product menu component:
<img width="400" alt="Screenshot 2022-09-28 at 15 49 09" src="https://user-images.githubusercontent.com/107254671/192796218-7f7e0c51-0155-4894-af33-4e51caff6ecb.png">

The [id].js file:

<img width="400" alt="Screenshot 2022-09-28 at 16 09 50" src="https://user-images.githubusercontent.com/107254671/192801681-75dd4885-09ae-49bd-9f39-e1e21a7746ff.png">
<img width="400" alt="Screenshot 2022-09-28 at 16 10 17" src="https://user-images.githubusercontent.com/107254671/192801689-f89ae582-dd56-4ffe-9389-00e2e120251f.png">

I haven't pushed this yet, hence the screenshots - let me know if you need any further info and thanks in advance! 🙏

**Comment by actyralf** [comment, marked as answer]
Could you please post a link to a pullrequest or a branch in which we can reproduce the behavior?

**Reply by barbigrade** [reply]
Thanks so much for the response! 
Oddly enough, the issue seems to have been resolved in the meantime - I realized that the double-mapping of the arrays was unnecessary and the data is now being rendered normally.
